local-storage/src/storage.rs

```diff
@@ -139,6 +141,29 @@ impl Storage {
         self.provider
             .get_dag_blocks_by_window(cid, offset, window_size)
     }
+
+    pub fn get_last_dag_cid(&self, cid: &str) -> Result<String> {
+        let dag_cids = self.get_all_dag_cids(cid)?;
```

Comment:
I feel like there should be a TODO to have the get_all* functions return an iterator rather than allocate. This is a good example of why.

Reply:
I think this could easily have been done inside of sqlite instead of returning all of the CIDs...but this function doesn't end up really getting used, so it can probably get removed.

But yes I agree that a good refactor would be returning iterator from those `get_all*` functions.